src/server.rs

```diff
@@ -308,4 +313,49 @@ mod tests {
 
         server.shutdown().await.unwrap();
     }
+
+    #[tokio::test]
+    async fn heartbeat() {
```

Comment:
This is a pretty big test, but it basically tests an entire lifecycle of the scuttlebutt server. So that should give at least some confidence things actually work properly now.

Reply:
yes. I am not against that kind of tests.